src/darwin/Framework/CHIP/MTRDeviceControllerDataStore.mm

```diff
@@ -495,83 +521,106 @@ - (void)_deleteAttributeValueForNodeID:(NSNumber *)nodeID endpointID:(NSNumber *
     return attributesToReturn;
 }
 
+#ifdef DEBUG
+- (void)unitTestPruneEmptyStoredAttributesBranches
+{
+    dispatch_sync(_storageDelegateQueue, ^{
+        [self _pruneEmptyStoredAttributesBranches];
+    });
+}
+#endif
+
 - (void)_pruneEmptyStoredAttributesBranches
 {
     dispatch_assert_queue(_storageDelegateQueue);
 
+    NSUInteger storeFailures = 0;
+
     // Fetch node index
-    NSMutableArray<NSNumber *> * nodeIndex = [self _fetchNodeIndex].mutableCopy;
-    NSUInteger nodeIndexCount = nodeIndex.count;
+    NSArray<NSNumber *> * nodeIndex = [self _fetchNodeIndex];
+    NSMutableArray<NSNumber *> * nodeIndexCopy = nodeIndex.mutableCopy;
 
-    NSUInteger storeFailures = 0;
     for (NSNumber * nodeID in nodeIndex) {
         // Fetch endpoint index
-        NSMutableArray<NSNumber *> * endpointIndex = [self _fetchEndpointIndexForNodeID:nodeID].mutableCopy;
-        NSUInteger endpointIndexCount = endpointIndex.count;
+        NSArray<NSNumber *> * endpointIndex = [self _fetchEndpointIndexForNodeID:nodeID];
+        NSMutableArray<NSNumber *> * endpointIndexCopy = endpointIndex.mutableCopy;
 
         for (NSNumber * endpointID in endpointIndex) {
             // Fetch endpoint index
-            NSMutableArray<NSNumber *> * clusterIndex = [self _fetchClusterIndexForNodeID:nodeID endpointID:endpointID].mutableCopy;
-            NSUInteger clusterIndexCount = clusterIndex.count;
+            NSArray<NSNumber *> * clusterIndex = [self _fetchClusterIndexForNodeID:nodeID endpointID:endpointID];
+            NSMutableArray<NSNumber *> * clusterIndexCopy = clusterIndex.mutableCopy;
 
             for (NSNumber * clusterID in clusterIndex) {
                 // Fetch endpoint index
-                NSMutableArray<NSNumber *> * attributeIndex = [self _fetchAttributeIndexForNodeID:nodeID endpointID:endpointID clusterID:clusterID].mutableCopy;
-                NSUInteger attributeIndexCount = attributeIndex.count;
+                NSArray<NSNumber *> * attributeIndex = [self _fetchAttributeIndexForNodeID:nodeID endpointID:endpointID clusterID:clusterID];
+                NSMutableArray<NSNumber *> * attributeIndexCopy = attributeIndex.mutableCopy;
```

Comment:
```suggestion
                NSMutableArray<NSNumber *> * attributeIndexCopy = [attributeIndex mutableCopy];
```
`mutableCopy` is not semantically a property (the reason for calling it is to get a new, mutable copy), so it should really not be called using property syntax. (Same for all the other instances)